core/templates/core/widget_tag_editor.html

```diff
@@ -0,0 +1,233 @@
+<div class="form_control" data-tags-url="{% url 'api:api-root' %}tags/"
+    {% for k, v in widget.attrs.items %}
+        {{ k }}={{ v }}
+    {% endfor %}>
+    {% csrf_token %}
+    <span class="prototype-tag btn badge badge-pill cursor-pointer mr-1" style="display: none;">
+        PROTOTYPE
+        <span class="add-remove-icon pl-1 pr-1">+ or -</span>
+    </span>
+    <div class="current_tags" style="min-height: 2em;">
+        {% for t in widget.value %}
+        <span data-value="{{ t.name }}" data-color="{{ t.color }}" class="tag btn badge badge-pill cursor-pointer mr-1" style="background-color: {{ t.color }};">
+            {{ t.name }}
+            <span class="add-remove-icon pl-1 pr-1">-</span>
+        </span>
+        {% endfor %}
+    </div>
+    <div class="new-tags">
+        <div class="create-tag-inputs input-group">
+            <input class="form-control" type="text" name="" placeholder="Tag name">
+            <div class="input-group-append">
+                <button class="btn btn-outline-primary bg-dark btn-add-new-tag" type="button">Add</button>
+            </div>
+        </div>
+        <span>Recently used:</span>
+        {% for t in widget.tag_suggestions.quick %}
+        <span data-value="{{ t.name }}" data-color="{{ t.color }}" class="tag btn badge badge-pill cursor-pointer mr-1" style="background-color: {{ t.color }};">
+            {{ t.name }}
+            <span class="add-remove-icon pl-1 pr-1">+</span>
+        </span>
+        {% endfor %}
+    </div>
+    <input 
+        type="hidden"
+        name="tags"
+        value="{% for t in widget.value %}&quot;{{ t.name }}&quot;{% if not forloop.last %},{% endif %}{% endfor %}"
+    >
+<script>
+        window.addEventListener('load', () => {
```

Comment:
This should be in its own `*.js` file. First version is inlined for easy of prototyping.

Reply:
I'll save some review for once we get this a little further along (I'm also just in Python brain right now and don't feel like switching to JS 😃)

Reply:
Yes, please do not review the js file yet! That thing is yucky right now... very much prototype-stuff. I wanted to show off how everything looks & feels with this draft and what changes to the python-code are necessary. Frontend-stuff typically is less likely to leave traces in the codebase for the long-haul imo, so I did not put much effort in the js-stuff yet.